src/Chapter4.hs

```diff
@@ -628,8 +662,11 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
+
+
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM a b = (fmap (\x-> (&&) x) a) <*> b
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the 
```
Just False `andM` _
``` 
should be `Just False` 🙂 

With the `Applicative` constraint, it is not possible to make it lazy. That is why we suggest using `Monad` 🙂 
But in reality, you only use `Applicative in here 🙂

Reply:
Noted. Thank you!